.github/internal/plans/docs/08-auth-guards.md

```
@@ -0,0 +1,134 @@
+// Usage
+router.Route("/admin", func(r chi.Router) {
+    r.Use(AuthMiddleware)
+    r.Use(RequireRole("admin"))
+    r.Get("/dashboard", adminDashboard)
+})
+```
+
+## Context Helpers
+
+See the [Context Helpers](context-helpers.md) guide for typed context key patterns.
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)